#!/usr/bin/env python
PACKAGE = "navex"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("width", double_t, 0, "Map width in meters"  , 3.0, 0.1, 10000.0)
gen.add("height", double_t, 0, "Map height in meters", 3.0, 0.1, 10000.0)
gen.add("resolution", double_t, 0, "Map resolution", 0.025, 0.01, 5.0)
gen.add("inflation_radius", double_t, 0, "Inflation radius in meters", 0.10, 0.0, 5.0)

gen.add("inflation_gradient_sig", double_t, 0, "Inflation gradient sigmma value of gaussian", 0.5, 0.0, 10)
gen.add("inflation_gradient_pow", double_t, 0, "Inflation gradient power", 1.0, 0.0, 16)

gen.add("robot_radius", double_t, 0, "Robot's radius in meters", 0.05, 0.0, 5.0)
gen.add("frame_id", str_t, 0, "Frame id of the map", "base_link")

exit(gen.generate(PACKAGE, "navex", "CostMap"))